common/sshtools.py

```diff
@@ -1309,3 +1309,27 @@ def writeKnownHostsFile(key):
     with open(knownHostFile, 'at') as f:
         logger.info('Write host key to {}'.format(knownHostFile))
         f.write(key + '\n')
+
+
+def select_private_key_file(files: list[Path]):
```

Comment:
Because you hand over a generator (Path.iterdir()) instead of a list.
```suggestion
def select_private_key_file(files: Iterable[Path]):
```

On top of the file: `from typing import Iterable`.